### ISSUE TITLE
Android: Collection Name not supported by Flatlist, SectionList, VirtualizedList, or ScrollView

### ISSUE THREAD

**Issue by amarlette** [facebook/react-native#30972]
## Description
Collections on Android have a name defined by their contentDescription property. This name will be announced as part of the "in list" and "out of list" announcements. Since we do not support the list/grid/pager roles, the contentDescription's set on these components don't announce these names upon focus into or out of any descendant elements.

Collection Name not from collection Not supported by:

- [ ] Flatlist #30968
- [ ] SectionList #30969
- [ ] VirtualizedList  #30970
- [ ] ScrollView  #30971

## React Native version:
v0.63

## Expected Behavior
If one of the list components has a contentDescription set on it, on first focus into that list, screen readers will announce "in list <content description>".  Upon focus out of the list, it should announce "out of list <content description>".

## Android Details
This should work properly if any of the collection roles were able to be set, which makes this task dependent on #30839.  By adding these roles, and properly setting the contentDescription the collection name should automatically be announced.

**Comment by sidverma32** [comment]
Hi @amarlette , I would like to work on this. Please guide me with some initial steps on "collection roles" to proceed, that would be helpful.

**Comment by fabriziobertoglio1987** [comment]
Related https://github.com/facebook/react-native/pull/33180

**Comment by fabriziobertoglio1987** [comment]
- [ ] verify if new prop/type is required for this API
- [ ] add setter in Java API to set the prop
- [ ] add JavaScript API
- [ ] record test cases and publish pr

**Comment by fabriziobertoglio1987** [comment]
task tracked on https://github.com/facebook/react-native/issues/30861#issuecomment-1193687248
